src/main/java/com/questhelper/helpers/quests/theheartofdarkness/TheHeartOfDarkness.java

```diff
@@ -0,0 +1,1028 @@
+        goDoFirstChallenge.addStep(inFirstTrialRoom, pickpocketAscended);
+        steps.put(32, goDoFirstChallenge);
+
+        ConditionalStep goDoSecondChallenge = new ConditionalStep(this, climbUpToFirstTrial);
+        goDoSecondChallenge.addStep(and(inSecondTrialRoom, combatStarted), completeCombatTrial);
+        goDoSecondChallenge.addStep(inSecondTrialRoom, startCombatTrial);
+        goDoSecondChallenge.addStep(inFirstTrialRoom, tellJanusPasscode);
+        steps.put(34, goDoSecondChallenge);
+        steps.put(36, goDoSecondChallenge);
+
+        ConditionalStep goStartThirdChallenge = new ConditionalStep(this, climbUpToFirstTrial);
+        goStartThirdChallenge.addStep(inSecondTrialRoom, talkToJanusAfterTrial);
+        steps.put(38, goStartThirdChallenge);
+
+        ConditionalStep goDoThirdChallenge = new ConditionalStep(this, climbUpToFirstTrial);
+        goDoThirdChallenge.addStep(and(inSecondTrialRoom, startedInvestigation, not(hasTalkedToTenoch)), talkToTenoch);
+        goDoThirdChallenge.addStep(and(inSecondTrialRoom, startedInvestigation, not(hasTalkedToSilia)), talkToSilia);
+        goDoThirdChallenge.addStep(and(inSecondTrialRoom, startedInvestigation, not(hasTalkedToAdrius)), talkToAdrius);
+        goDoThirdChallenge.addStep(and(inSecondTrialRoom, startedInvestigation, not(hasTalkedToEleuia)), talkToEleuia);
+        goDoThirdChallenge.addStep(and(inSecondTrialRoom, startedInvestigation, tenochGuilty), accuseTenoch);
+        goDoThirdChallenge.addStep(and(inSecondTrialRoom, startedInvestigation, siliaGuilty), accuseSilia);
+//        goDoThirdChallenge.addStep(and(inSecondTrialRoom, startedInvestigation, adriusGuilty), accuseAdrius);
+        goDoThirdChallenge.addStep(and(inSecondTrialRoom, startedInvestigation, eleuiaGuilty), accuseEleuia);
+        goDoThirdChallenge.addStep(and(inSecondTrialRoom, startedInvestigation), talkToTenoch);
+        goDoThirdChallenge.addStep(inSecondTrialRoom, talkToPrinceToStartThirdTrial);
+        steps.put(40, goDoThirdChallenge);
+
+        ConditionalStep goToFourthChallenge = new ConditionalStep(this, climbUpToFirstTrial);
+        goToFourthChallenge.addStep(inSecondTrialRoom, goUpToFinalTrial);
+        steps.put(42, goToFourthChallenge);
+
+        ConditionalStep goDoFourthChallenge = new ConditionalStep(this, climbUpToFirstTrial);
+        goDoFourthChallenge.addStep(inSecondTrialRoom, fightPrince);
+        steps.put(44, goDoFourthChallenge);
+
+        ConditionalStep wrapUpFourthChallenge = new ConditionalStep(this, talkToJanusAfterAllTrials);
+        wrapUpFourthChallenge.addStep(inSecondTrialRoom, talkToJanusAfterPrinceFight);
+        steps.put(46, wrapUpFourthChallenge);
+
+        steps.put(48, searchChestForEmissaryRobes);
+
+        ConditionalStep goToTemple = new ConditionalStep(this, talkToItzlaToFollow);
+        goToTemple.addStep(and(princeIsFollowing, emissaryRobes), enterTemple);
```

Comment:
While watching the cutscene here, it activated the "talkToItzlaToFollow" step. Could add a check if we're in a cutscene to be on the enterTemple step. something like `and(or(princeIsFollowing, inCutscene), emissaryRobes)`